llvm/lib/Target/RISCV/RISCVInstrInfoXRivos.td

```diff
@@ -25,3 +65,27 @@ defm RI_VZIP2B_V : VALU_IV_V<"ri.vzip2b", 0b010100>;
 defm RI_VUNZIP2A_V : VALU_IV_V<"ri.vunzip2a", 0b001000>;
 defm RI_VUNZIP2B_V : VALU_IV_V<"ri.vunzip2b", 0b011000>;
 }
+
+//===----------------------------------------------------------------------===//
+// XRivosVisni
+//===----------------------------------------------------------------------===//
+
+let Predicates = [HasVendorXRivosVisni], DecoderNamespace = "XRivos",
+  Inst<6-0> = OPC_CUSTOM_2.Value, mayLoad = false, mayStore = false,
+  hasSideEffects = false in {
+
+let isReMaterializable = 1, isAsCheapAsAMove = 1, vm = 0, vs2=0 in
+def RI_VZERO : RVInstV<0b000000, 0b00000, OPCFG, (outs VR:$vd),
+                     (ins), "ri.vzero", "$vd">;
```

Comment:
```suggestion
                     (ins), "ri.vzero.v", "$vd">;
```
The document uses this mnemonic - should we use it here? I'm not entirely up to date on the V-extensions, and how mnemonics are generally handled there though, and what the conventions are.

Reply:
Fixed, thanks.  I copied this from a downstream change from before we'd finalized the naming in that spec.